ci_framework/roles/ci_network/tasks/apply.yml

```diff
@@ -0,0 +1,43 @@
+---
+- name: Assert network configuration is valid
+  block:
+    - name: VLAN data are present
+      when:
+        - network.key is match('^vlan.*')
+      ansible.builtin.assert:
+        that:
+          - network.value.parent_iface is defined
+        msg: "parent_iface is needed for VLAN interface"
+
+    - name: IPv4 is defined
+      ansible.builtin.assert:
+        that:
+          - network.value.ip4 is defined
+        msg: "ip4 is needed for any interface"
+
+- name: "Apply network {{ network.key }}"
+  become: true
+  when:
+    - network.value.config_nm | default(true) | bool
+  community.general.nmcli:
+    autoconnect: true
+    state: "{{ network.value.state | default('present') }}"
+    conn_name: "{{ network.value.connection | default(network.key) }}"
+    ifname: "{{ network.value.iface }}"
+    type: "{{ network.key is match('^vlan.*') | ternary('vlan', 'ethernet') }}"
```

Comment:
Why should VLAN networks start with the vlan prefix? Sounds like a limitation. I think that a better criteria is to just check if `network.value.vlan` is defined, and if it is, assume it's a vlan interface.

Reply:
let's assume ppl know what they do and let's check if "vlan" is set.